### PR TITLE
added tests for caclulus_kit

### LINKED ISSUE #116: Add unit tests for calculus_kit.py
Create unit tests for calculus_kit.py, then add unit tests covering its public API (gradients, jacobians, edge cases, numeric stability). Deferred from PR tests/forecasting-tests-78.

This only needs to check delegation and nothing else (since unit tests for each object is being done in separate files/tests. In other wirds this just checks if we are truly calling build_hessian, build_jacobian and build_gradient.

### PR DESCRIPTION
**Pull request by nikosarcevic**
Adds delegation tests for CalculusKit.
Tests ensure that gradient, jacobian, hessian, and hessian_diag correctly forward calls, arguments, and return values to the underlying calculus builders.

This closes #116

### CODE REVIEW

File: tests/test_caclulus_kit.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+
+def test_calculuskit_stores_function_and_x0():
+    """Tests that CalculusKit stores function and x0 correctly."""
+    x0 = [0.1, 0.2]
+    ck = CalculusKit(scalar_function, x0)
+
+    assert ck.function is scalar_function
+    assert isinstance(ck.x0, np.ndarray)
+    assert ck.x0.dtype == float
```
Comment (lonbar) [inline tests/test_caclulus_kit.py]: Also, there is no real need for `x0` to be a `float`. Perhaps it's better to check against the type of the input `x0`?

File: tests/test_caclulus_kit.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        recorder,
+        raising=True,
+    )
+
+    x0 = [0.5, 1.5]
+    ck = CalculusKit(scalar_function, x0)
+
+    out = ck.gradient("extra", method="adaptive", n_workers=3)
+
+    # Delegation checks
```
Comment (lonbar) [inline tests/test_caclulus_kit.py]: Please don't write comments that describe the code; write comments to give further details or context that is missing in the code (as described in the [Google style guide](https://google.github.io/styleguide/pyguide.html#385-block-and-inline-comments)).

File: tests/test_caclulus_kit.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    assert recorder.func is scalar_function
+    np.testing.assert_allclose(recorder.x0, np.asarray(x0, dtype=float))
+    assert recorder.args == ("extra",)
+    assert recorder.kwargs == {"method": "adaptive", "n_workers": 3}
+
+    # Return passthrough
```
Comment (lonbar) [inline tests/test_caclulus_kit.py]: Please don't write comments that describe the code; write comments to give further details or context that is missing in the code (as described in the [Google style guide](https://google.github.io/styleguide/pyguide.html#385-block-and-inline-comments)).

File: tests/test_caclulus_kit.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        recorder,
+        raising=True,
+    )
+
+    x0 = np.array([1.0, 2.0])
+    ck = CalculusKit(vector_function, x0)
+
+    out = ck.jacobian(foo="bar")
+
+    assert recorder.func is vector_function
```
Comment (lonbar) [inline tests/test_caclulus_kit.py]: THe questions I asked above apply here too.

File: tests/test_caclulus_kit.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        # sentinel Hessian diagonal
+        return np.array([9.0, 8.0])
+
+
+def test_calculuskit_stores_function_and_x0():
+    """Tests that CalculusKit stores function and x0 correctly."""
+    x0 = [0.1, 0.2]
+    ck = CalculusKit(scalar_function, x0)
+
+    assert ck.function is scalar_function
```
[the remark this answers is not in the document]
Comment (nikosarcevic) [reply tests/test_caclulus_kit.py]: yeah here it’s just an arbitrary callable --- the test only checks that CalculusKit stores the function and x0 correctly. there’s no requriement that the function be scalar-valued in this test

i have fixed the docstring in fcc82a70e94968a7d71874a2a69bd115dd285fe2

File: tests/test_caclulus_kit.py (hunk 3 above)
Comment (nikosarcevic) [reply tests/test_caclulus_kit.py]: removed in faca3173e9048c4d9b06f6ce643c1c4c80fd2dd5
